### PR TITLE
Fix some msvc specific warnings

### PR DESCRIPTION
**Pull request by stinos**
(I also just noticed github goes berserk on the `*/*const*/` :)

### CODE REVIEW

File: py/mpconfig.h [hunk 1]
```diff
@@ -739,4 +739,8 @@ typedef double mp_float_t;
 #define MP_UNLIKELY(x) __builtin_expect((x), 0)
 #endif
 
+// Use to refer to extern structs of which actual type is not known.
+// For instance 'extern struct _dummy_t mp_sys_stdin_obj;'
+struct _dummy_t;
```
Comment (dpgeorge) [inline py/mpconfig.h]: It's not really a configuration option.  I'd put it in misc.h.  And maybe name it _mp_dummy_t.

Comment (stinos) [inline py/mpconfig.h]: I just figured it is not needed to define this dummy struct at all: `extern struct _mp_dummy_t mp_sys_stdin_obj;` is fine as-is and the compiler doesn't need to know more to be able to make a pointer from it.

We could still declare _mp_dummy_t in misc.h but it does not do anything, as in: compiler won't warn you if you make typeos like ```extern struct _mp_oops_dummy_t``` so in the end it only adds to maintainance and serves no purpose. Unless there are compilers which do not handle it but I doubt that?

Comment (dpgeorge) [inline py/mpconfig.h]: I agree we don't need the declaration in a header file.  Even gcc with -Wpedandic doesn't complain about it.

I would just name it `struct _mp_dummy_t` to be consistent with other names, and highlight the fact that it doesn't matter what it is and that a given port might have it as one type, another port as another type.

**Comment by dpgeorge** [comment]
Thanks!

**Pull request merged**